### PR TITLE
config(eslint): allow test files to import from 'internal/test*'

### PR DESCRIPTION
**Pull request by DLehenbauer**
Avoid need to suppress 'import/no-intenal-modules' when importing from:
- `/test*`
- `/internal/test*`
with expectation is that `test-utils` (public) and `internal/test-utils` (internal) will most often be the exports for test utilities.

Additionally removes exception for `/beta` imports as they are not expected internally.

### CODE REVIEW

File: common/build/eslint-config-fluid/minimal-deprecated.js [hunk 1]
```diff
@@ -3,6 +3,21 @@
  * Licensed under the MIT License.
  */
 
+/**
+ * Shared list of permitted imports for configuring and override the `import/no-internal-modules` rule.
+ */
+const permittedImports = [
+	// Within Fluid Framework, allow import of '/internal', '/beta', and 'alpha' exports.
+	"@fluidframework/*/beta",
+	"@fluidframework/*/alpha",
```
Comment (jason-ha) [inline common/build/eslint-config-fluid/minimal-deprecated.js:12]: I am pretty sure we do not want alpha imports in our codebase except for special cases.
I am not sure about utility of beta.
Internally I think we want public and internal in almost all cases.

**Review by jason-ha — CHANGES_REQUESTED** [review]

**Comment by jason-ha** [comment]
bump minor package.json version
Assuming this doesn't beat #19901 to be merged, there would be conflict.

**Review by jason-ha — APPROVED** [review]

File: common/build/eslint-config-fluid/minimal-deprecated.js [hunk 2]
```diff
@@ -430,6 +434,14 @@ module.exports = {
 				// Disabled for test files
 				"@typescript-eslint/consistent-type-exports": "off",
 				"@typescript-eslint/consistent-type-imports": "off",
+
+				// For test files only, additionally allow import of '/test' exports.
+				"import/no-internal-modules": [
+					"error",
+					{
+						allow: ["@fluidframework/*/test"].concat(permittedImports),
```
Comment (jason-ha) [inline common/build/eslint-config-fluid/minimal-deprecated.js:442]: I am now thinking we should make this more relaxed.
"@fluid*/*/test*" and "@fluid*/*/internal/test*"
Seem good. I do think "test-utils" is a better name than "test" in most cases because we aren't _normally_ testing that package.

**Review by jason-ha — CHANGES_REQUESTED** [review]
(back to block since you say you may not notice comments otherwise)
And you can bump the minor this PR too.

**Comment by DLehenbauer** [comment]
> (back to block since you say you may not notice comments otherwise) And you can bump the minor this PR too.

Blocking is appreciated while we're working on node16/trimming.  (A couple of times a day, I open all the pending PRs in browser tabs and quickly scan through them to see what needs to happen to nudge them forward.)

File: common/build/eslint-config-fluid/minimal-deprecated.js [hunk 3]
```diff
@@ -7,8 +7,8 @@
  * Shared list of permitted imports for configuring and override the `import/no-internal-modules` rule.
  */
 const permittedImports = [
-	// Within Fluid Framework allow import of '/internal'.
-	"@fluidframework/*/internal",
+	// Within Fluid Framework allow import of '/internal' from other FF packages.
+	"@fluid*/*/internal",
```
Comment (jason-ha) [inline common/build/eslint-config-fluid/minimal-deprecated.js:12]: I did think we wanted to leave this just to @fluidframework. We don't need (AFAIK), to apply trimming to other scopes. If those cases exist they are rare and can be case-by-case allowed.

**Review by jason-ha — CHANGES_REQUESTED** [review]
Description will need updated. Also I see it has a typo in "no-intenal-modules".

File: build-tools/lerna.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-	"version": "0.35.0",
+	"version": "0.36.0",
```
Comment (tylerbutler) [inline build-tools/lerna.json:2]: Unrelated changes?

**Review by jason-ha — APPROVED** [review]
